cli/resolver/tag.go

```diff
@@ -1,11 +1,103 @@
 package resolver
 
 import (
+	"fmt"
+	"net/url"
+	"strings"
+
 	"github.com/quintilesims/layer0/client"
+	"github.com/quintilesims/layer0/common/errors"
 )
 
-func NewTagResolver(client client.Client) ResolverFunc {
+func NewTagResolver(c client.Client) ResolverFunc {
 	return func(entityType, target string) ([]string, error) {
-		return []string{target}, nil
+		switch entityType {
+		case "environment", "job":
+			return resolveUnqualifiedEntity(c, entityType, target)
+		case "service", "load_balancer", "task":
+			targets := strings.Split(target, ":")
+			switch len(targets) {
+			case 1:
+				return resolveUnqualifiedEntity(c, entityType, target)
+			case 2:
+				return resolveFullyQualifiedEntity(c, entityType, targets[0], targets[1])
+			default:
+				return nil, fmt.Errorf("Invalid target format (expected ENVIRONMENT:%s)", strings.ToUpper(entityType))
+			}
+		case "deploy":
+			targets := strings.Split(target, ":")
+			switch len(targets) {
+			case 1:
+				return resolveUnqualifiedEntity(c, entityType, target)
```

Comment:
Do you not need an `resolveUnqualifiedDeploy` here as the `resolveUnqualifiedEntity` doesn't filter by deploy entity type?

Reply:
I'm not 100% sure I understand your question, but `resolveUnqualifiedEntity` doesn't do any filtering period. It just fuzzy matches based of the "raw" input and whatever `entityType` you pass in (in this case, we know it's `"deploy"`). 

Or maybe you are saying this should be: `return resolveUnqualifiedEntity(c, "deploy", target)`?

Reply:
Oh I see - upon reading this again, I misunderstood what the code was doing.